### PR TITLE
Issue #809 Update project dependencies

### LINKED ISSUE #809: Update project dependencies

### PR DESCRIPTION
**Pull request by Manangka**
Fixes #809

This commit updates the dependencies of the project.
There are some issues with the latest dask packag,s which isn;t playing nice with raterio, and with the pytest package, which has issues with pytest-cases. I've pinned both packages to a version that doesn't cause any issue.
Further there has been an update of black which results in a new formatting rule being applied.

Hopefully this is the last time we have to update manually as the pipeline script to do this automatically is almost done

# Checklist

- [x] Links to correct issue
- [ ] Update changelog, if changes affect users
- [x] PR title starts with ``Issue #nr``, e.g. ``Issue #737``
- [ ] Unit tests were added
- [ ] **If feature added**: Added/extended example

### CODE REVIEW

File: pixi.toml [hunk 1]
```diff
@@ -56,7 +56,7 @@ bottleneck = "*"
 cftime = ">=1"
 contextily = ">=1"
 cytoolz = "*"
-dask = "*"
+dask = "<2024.1.0" # Newer version incompatible with raterio
```
Comment (luitjansl) [inline pixi.toml:59]: typo in raterio

**Review by luitjansl — APPROVED** [review]
looks good

File: pixi.toml (hunk 1 above)
Comment (Manangka) [reply pixi.toml]: Fixed

**Review by JoerivanEngelen — APPROVED** [review]
Thanks for doing this! And for fixing the incompatibility things:

Relevant issue for package incompatibility:
* https://github.com/smarie/python-pytest-cases/issues/330

**Pull request merged**